src/KustoExpressionParser.ts

```diff
@@ -43,17 +43,25 @@ export class KustoExpressionParser {
     };
 
     const parts: string[] = [];
+    const isDynamic = query.search.property.name.includes(DYNAMIC_TYPE_ARRAY_DELIMITER);
+    const name: string = this.addExpanPartsdIfNeeded(query.search.property.name, parts);
     this.appendProperty(context, query.expression.from, parts);
     this.appendTimeFilter(context, undefined, parts, tableSchema);
 
-    if (query.index) {
+    //query.index is used by the legacy query editor
+    if (query.index && !isDynamic) {
       const where = replaceByIndex(query.index, query.expression.where, query.search);
       this.appendWhere(context, where, parts, 'where');
     }
-    const column = query.search.property.name;
+
+    if (isDynamic) {
+      parts.push(`mv-expand ${name} = ${parts[0]}`);
+      parts.push(`where ${query.search.operator.name}(${name})`);
+      parts.shift();
```

Comment:
Needed since the top of `parts` at this point will have the name of array, e.g. `Modes`.

Reply:
you need to do this because `addExpanPartsdIfNeeded` is meant to be called with an specific array for "parts to expand". If you define that array, you can use here the predefined function to add `mv-expand`.

Also, you can use the predefined function to `appendWhere` to do the condition. The only thing that you need to modify is the `search` query since it's still pointing to the "unexpanded" column name.